concepts/pointers/introduction.md

```diff
@@ -1,3 +1,149 @@
+```
+
+Assigning to `*pa` will change the value stored at the memory address `pa` holds. Since `pa` holds the memory address of `a`, by assigning to `*pa` we are effectively changing the value of `a`! 
+
+A note of caution however: always check if a pointer is not `nil` before dereferencing. Dereferencing a `nil` pointer will make the program panic at runtime!
+
+### Pointers to structs
+
+So far we've only seen pointers to primitive values. We can also create pointers for structs:
+
+```go
+type Person struct {
+    Name string
+    Age  int
+}
+
+var peter Person
+peter = Person{Name: "Peter", Age: 22}
+
+var p *Person
+p = &peter
+```
+
+We could've also created a new `Person` and immediately get a pointer to it:
+
+```go
+var p *Person
+p = &Person{Name: "Peter", Age: 22}
+```
+
+When we have a pointer to a struct, we don't need to dereference the pointer before accessing one of the fields:
+
+```go
+var p *Person
+p = &Person{Name: "Peter", Age: 22}
+
+fmt.Println(p.Name) // Output: "Peter" 
+                    // Go automatically dereferences 'person' to allow
+                    // access to the 'Name' field
+```
+
+## Slices and maps are already pointers
+
+Slices and maps are special types because they already have pointers in their implementation. This means that more often that not, we don't need to create pointers for these types to share the memory address for their values. Image we have a function that adds a key to a map:
+
+
+```go
+func addPeterAge(m map[string]int){
+	m["Peter"] = 22
+}
+```
+
+If we create a map and call this function, the changes the function made to the map persist after the function ended. This is a similar behavior we get if we were using a pointer, but note how on this example we are not using any referencing/dereferencing or any of the pointer syntax:
+
+```go
+ages := make(map[string]int)
+addPeterAge(ages)
+fmt.Println(ages)
+// Output: map[Peter:22]
+// The changes the function 'addPeterAge' made to the map are visible after the function ends!
+```
+
+## Why are pointers useful?
+
+Pointers are a way to share memory with other parts of our program. When you pass a pointer as an argument to a function, you are sharing the memory address of a value with that function. The function can then use that memory address to access and modify the value. All the modifications that the function makes to that value through the pointer are then visible to other parts of the program when the function ends. Thus, pointers can be a way to share memory.
+
+Another situation where sharing memory can be useful is to avoid copies of large amounts of data. If we have a piece of data that takes a lot of space in memory, it might not be practical to copy that data to every place we need it. In that situation, we might consider using a pointer and pass the pointer around instead. That way we avoid copies of large amounts of data, but we allow several parts of the program to access it via the pointer. 
```

Comment:
We can probably remove this from here now, as we have a summary version at the top. However, it'd be good to have this slightly more detailed version stay in the about.md

Reply:
I like this change a lot!

I talked a bit with some people to know their troubles learning pointers before doing this exercise. One of the things they told me is that they had trouble learning pointers because they were not seeing why they were useful. That's why I made this little section, but I agree 100% that having this upfront is even better.

Change made.